apps/dashboard/src/pages/environments.tsx

```diff
@@ -1,13 +1,29 @@
 import { PageMeta } from '@/components/page-meta';
+import { ApiServiceLevelEnum } from '@novu/shared';
 import { useEffect } from 'react';
 import { CreateEnvironmentButton } from '../components/create-environment-button';
 import { DashboardLayout } from '../components/dashboard-layout';
+import { FreeTierState } from '../components/environments-free-state';
 import { EnvironmentsList } from '../components/environments-list';
+import { useAuth } from '../context/auth/hooks';
+import { useFetchEnvironments } from '../context/environment/hooks';
+import { useFetchSubscription } from '../hooks/use-fetch-subscription';
 import { useTelemetry } from '../hooks/use-telemetry';
 import { TelemetryEvent } from '../utils/telemetry';
 
 export function EnvironmentsPage() {
+  const { currentOrganization } = useAuth();
+  const { environments = [], areEnvironmentsInitialLoading } = useFetchEnvironments({
+    organizationId: currentOrganization?._id,
+  });
   const track = useTelemetry();
+  const { subscription } = useFetchSubscription();
+
+  const isPaidTier =
+    subscription?.apiServiceLevel === ApiServiceLevelEnum.BUSINESS ||
+    subscription?.apiServiceLevel === ApiServiceLevelEnum.ENTERPRISE;
+  const isTrialActive = subscription?.trial?.isActive;
+  const canAccessEnvironments = areEnvironmentsInitialLoading || !subscription || (isPaidTier && !isTrialActive);
```

Comment:
```suggestion
  const isPaidTier =
    subscription?.apiServiceLevel !== ApiServiceLevelEnum.FREE && !subscription?.trial?.isActive;
  const canAccessEnvironments = isPaidTier;
```

can you elaborate why `canAccessEnvironments` contains `areEnvironmentsInitialLoading` and `subscription` flags? 
in case data is loading (areEnvironmentsInitialLoading) we need to render loading state and not another component (`<FreeTierState />`)

if there is not subscription (`!subscription`), the isPaidTier will be false as well.

Reply:
The probelm is that we have multiple loading states and I want to make sure that we don't have this flashing to the user,
- Environments loading
- Subscription loading

The loader is the environments list that I want to show until all data is listed.